pkg/api/v1/pod/util.go

```diff
@@ -50,38 +50,65 @@ func FindPort(pod *v1.Pod, svcPort *v1.ServicePort) (int, error) {
 	return 0, fmt.Errorf("no suitable port for manifest: %s", pod.UID)
 }
 
+// ContainerType signifies container type
+type ContainerType int
+
+// DefaultContainers defines default behavior: Iterate containers based on feature gates
+const DefaultContainers ContainerType = 0
+
+const (
+	// Containers is for normal containers
+	Containers ContainerType = 1 << iota
+	// InitContainers is for init containers
+	InitContainers
+	// EphemeralContainers is for ephemeral containers
+	EphemeralContainers
+)
+
+// AllContainers specifies that all containers be visited
+const AllContainers ContainerType = (InitContainers | Containers | EphemeralContainers)
```

Comment:
```
const (
	Containers ContainerType = 1 << iota
	InitContainers
	EphemeralContainers
	// must be final element in this block
	lastContainerType
)

const AllContainers ContainerType = lastContainerType - 1
```

Would work, although I expect that adding new types is going to be rare. Feel free to ignore.